src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1604,20 +1604,32 @@ public boolean resumeBootstrap()
                 @Override
                 public void onSuccess(StreamState streamState)
                 {
-                    bootstrapFinished();
-                    if(isSurveyMode)
+                    try
                     {
-                        logger.info("Startup complete, but write survey mode is active, not becoming an active ring member. Use JMX (StorageService->joinRing()) to finalize ring joining.");
+                        bootstrapFinished();
+                        // start participating in the ring.
+                        // pretend we are in survey mode so we can use joinRing() here
+                        if (isSurveyMode)
+                        {
+                            logger.info("Startup complete, but write survey mode is active, not becoming an active ring member. Use JMX (StorageService->joinRing()) to finalize ring joining.");
+                        }
+                        else
+                        {
+                            isSurveyMode = false;
```

Comment:
This assignment is superfluous, but it wasn't introduced by this patch at least.